---
title: "Using RNAseq data to analyze gene expression in *Haemophilus parainfluenzae* aerobically and anaerobically "
author: "Thaís Palma"
date: "May 13, 2019"
output: pdf_document  
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.width = 6, fig.height = 3, fig.align = 'center')
#load packages
library(tidyverse)
library(grid)

#load data
genes = read.csv("hpara_genes.csv")

```
# Introduction

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The oral cavity harbors a very dynamic and complex microbial community. More than 700 species of microorganism colonize different niches in the mouth. Among them, one of the most abundant species is *Haemophilus parainfluenzae (H. para*). *H. para* is a Gram-negative, facultatively anaerobic coccobacillus. Even though it is a commensal bacterium in the oral cavity, in some cases it can be associated with local diseases, such as gingivitis (Park et al, 2015), or systemic diseases such as endocarditis. It is estimated that *H. para* is responsible for approximately 3% of the infective endocarditis cases (Posse et al, 2018). Despite its prevalence, *H. para* is not very well studied and more research is necessary to better understand its metabolism, and its genes expressions under different circumstances associated with its opportunistic pathogenicity characteristic. Based on that, this paper aims to investigate the difference in genes expression of *H. para* under aerobic and anaerobic conditions using an RNAseq database. Below I describe the data and report the initial results of my study. Out of 1764 observations generated by RNAseq, 258 had significantly different levels of genes expression (88 were upregulated under anaerobic conditions and 170 under aerobic conditions). Out of the 258 significant results, 48 (27 anaerobic, and 21 aerobic) presented 'fold change' equal or higher than 2.5. Future research will analyze each one of the 48 genes and, for some of them, mutant strains will be built and tested under different conditions.  

# Methods

##Strain, growth condtion and RNA extraction

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The sample *Haemophilus parainfluenzae* (ATCC 33392) was used to perform the RNAseq analyses. The sample was streaked on brain heart infusion (BHI) plates and incubated in 5% CO2 for 24 hours at 37oC. After that, one isolated colony was transfered to 3 ml of BHI liquid and incubated either aerobically or anaerobically, using a candle jar for 24 hours at 37oC. Next, 20 $\mu$l of culture was spoted on a membrane and grew for 24h under the same conditions as mentioned on the previous step. Finally, the membrane was washed with Trizol to avoid RNA degradation and frozen (-80oC). The samples were sent to have the RNA extracted and sequenced.

##RNA analyses
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The raw data was processed and analyzed using HTSeq to count the reads and Bowtie to align them. After that, I used an R script (available on github) to generate the table that is the base to this paper. Statistical analyzes, graphs, and tables were generated using Rstudio.

##R analyses

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The first step was to organize the table. To do so, I used the function "rename" to rename the columns and label the unlabeled ones. Next, I filtered all the cells that did not have value specified with the function "filter". The function "mutate" was used to create new variables that were not in the original table. For instance, I created a column named condition to specify whether the samples were incubated under aerobic or anaerobic condition. 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;To differentiate gene expression under the conditions aerobic and anaerobic, I considered and filtered the values of 'fold change'. This variable compares how many times (fold) the genes expression is smaller or larger in the two conditions. For instance, when the fold change was 1, that means that there is no difference in genes expression between the two conditions. Values larger than 1 indicate higher expression in anaerobic conditions, while values smaller than 1 indicated higher expression aerobically.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Given the large number of observations generated originally (1764 observation), I concentrate only on the ones that have significant fold change, i.e., where there is a significant difference in gene expression between aerobic and anaerobic conditions. To do so, I filtered for p-values smaller than 0.05 and created a new table (genes_filter). The new table also has the information specifying the condition (aerobic or anaerobic) as well as the name of the genes, proteins, and fold change

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;After that, for each condition (aerobic and anaerobic) I analyzed the observations that had fold change higher or equal to 2.5 and generated two tables. Table 1 and Table 2 present not only fold change values but also the protein that is codified by the genes that are upregulated. The codes below show these operations.


```{r start, tidy=TRUE, tidy.opts=list(width.cutoff=6)}

# Rename variables
genes = genes %>% rename(parent = X, name_wp = X.1, inference = X.2, protein = X.3, 
                         trans_table = X.4)

# Filtering out no values
genes = genes%>% filter(fold.change != "#VALUE!") %>% mutate(fold.change = as.numeric
                                                             (as.character(fold.change)))

# Create new variables
genes = genes %>% mutate(sig = ifelse(padj >= 0.05, "Not significant","Significant"))

#Convert aerobically to same scale as anaerob.
genes = genes %>% mutate (fold_change_scaled = 
                            ifelse (fold.change < 1, 1/fold.change, fold.change))
write.csv(genes, file = "gene.csv")

# Filter for conditions
genes_filter = genes%>% filter(sig == "Significant", 
                               gene_description != "product=hypothetical protein") 
write.csv(genes_filter, file = "genes_filter.csv")  
genes_filter = genes_filter %>% mutate(condition = ifelse(fold.change > 1, "anaero", "aero"))
``` 

# Results

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Figure 1 represents the distribution of fold change in gene expression of *H. para* aerobically and anaerobically for all genes. The dashed line represents the median. This figure gives a general idea of the distribution of differences in genes expression across conditions, however, it is hard to interpret given the difference in scale. Also, it includes many tests that are not statically significant (note how the distribution is centered around 1, meaning no change). To better visualize the changes in gene expression between anaerobic and aerobic, first I separated the tests that were statistically significant (p-value < 0.05). Figure 2 shows both, significant and not significant fold change distribution in gene expression.

```{r genes, echo=FALSE}
ggplot (genes, aes (x = fold.change)) +
  geom_density () +
  theme_minimal() +
  geom_vline(xintercept = median(genes$fold.change), linetype = 'dashed', color = 'darkblue') +
  scale_x_continuous(breaks = c(0,round(median(genes$fold.change),2),5,10,15)) +
  theme(panel.grid.minor = element_blank()) +
  labs(x = 'Fold change', y = 'Density', caption = ('Fig.1: Distribution of fold change in gene expression.'))

```



```{r sig , echo=FALSE, warning=FALSE, message=FALSE}
genes = genes %>% mutate(sig = ifelse(padj >= 0.05, "Not significant","Significant"))

ggplot (genes %>% filter(sig != "NA"), aes (x = fold.change, color = sig, fill = sig)) +
  scale_x_continuous(breaks = seq(0,15,1))+
  geom_density (alpha = 0.2) +
  theme_minimal() +
  theme(legend.title = element_blank(), panel.grid.minor = element_blank(), plot.margin = margin(0,0,0,0,'cm')) +
  labs(x = 'Fold change', y = 'Density', caption = expression('Fig.2: Fold change in gene expresion in'~italic('Haemophilus parainfluenzae')))

```

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Analyzing Figure 2 it is possible to see that when the "not significant" fold changes are excluded, there are many cases of actual change, in which fold changes are either clearly smaller or larger than one. 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The separation between statistically significant and not significant tests around the 0.05 threshold could be misleading if there were many tests in which the p-value was close but above 0.05. Figure 3 shows the distribution of p-values for all tests, providing reassurance that this is not the case. Figure 3 indicates that out of 1764 expressed genes, 275 (15.6%) are statistically significant.  
   

```{r pval , echo=FALSE, warning=FALSE, message=FALSE}

ggplot (genes, aes (x = padj, fill = sig)) +
  geom_histogram (binwidth = 0.02, position = 'dodge') +
  theme_minimal() +
  geom_vline(xintercept = 0.05, linetype = 'dashed', color = 'darkblue') +
  scale_x_continuous(breaks = c(0,0.05, seq(0.2, 1, 0.2)), limits = c(0,1)) +
  theme(panel.grid.minor = element_blank())+
  labs(x = 'p-value', y = 'Number of genes', caption = ('Fig.3: Distribution of p-values over the genes.'))


```

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The difference between aerobic and anaerobic genes expression is represented in Figure 4. In this plot only significant fold changes (p < 0.05) were considered. According to this analysis, it is possible to affirm that under aerobic (aero) conditions, the fold change varies from 1.35 to 5. On the other hand, under anaerobic (anaero) conditions, the fold change is more spread and varies from 1.35 to 10. However, it is important to note that large part of the genes is concentrated around 2-fold change. 

```{r aero_anaero , echo=FALSE, warning=FALSE, message=FALSE}
ggplot (genes_filter, aes (x = fold_change_scaled, color = condition, fill = condition)) +
  geom_density (alpha = 0.2) +
  theme_minimal() +
    geom_vline(xintercept = median(genes_filter$fold_change_scaled[genes_filter$condition == 'aero']), linetype = 'dashed', color = 'red') +
  geom_vline(xintercept = median(genes_filter$fold_change_scaled[genes_filter$condition == 'anaero']), linetype = 'dashed', color = '#339999') +
  theme(panel.grid.minor = element_blank()) +
  labs(x = 'Fold change', y = 'Density', caption = ('Fig.4: Difference in the expression of genes.')) +
  annotate('text', x = 1.4, y = 1, label=(round(median(genes_filter$fold_change_scaled[genes_filter$condition=='aero']),2)), color = 'red' ) +
  annotate('text', x = 2.5, y = 0.7, label=(round(median(genes_filter$fold_change_scaled[genes_filter$condition=='anaero']),2)), color = '#339999' )
```

```{r aero_anaero_large , echo=FALSE, warning=FALSE, message=FALSE}
genes_filter_large_aero = genes_filter %>% filter(sig == "Significant", gene_description != "product=hypothetical protein", condition == 'aero', fold_change_scaled >= 2.5)
write.csv(genes_filter_large_aero, file = "genes_filter_large_aero_raw.csv")

genes_filter_large_anaero = genes_filter %>% filter(sig == "Significant", gene_description != "product=hypothetical protein", condition == 'anaero', fold_change_scaled >= 2.5)
write.csv(genes_filter_large_anaero, file = "genes_filter_large_anaero_raw.csv")
```

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Due to the large number of observations created by the RNAseq analyses, I decided to analyze the cases in which the upregulated genes were the most different across the two conditions. To do so, I filtered the results based on the p-value (< 0.05), and extreme fold change (larger than 2.5).


```{r genes_large_aero , echo=FALSE, warning=FALSE, message=FALSE}
library(knitr)

genes_large_aero = read.csv('genes_filter_large_aero.csv')
genes_large_aero = genes_large_aero %>% select(gene_description, fold_change_scaled) %>% mutate(fold_change_scaled = round(fold_change_scaled, 1)) %>% arrange(desc(fold_change_scaled)) %>% rename('Fold change'= fold_change_scaled, 'Protein' = gene_description) 

kable(genes_large_aero, caption = ' Proteins coded by uperregulated genes and fold change (>= 2.5) in gene expression under aerobic condition')

```

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Table 1 shows proteins coded by upregulated genes under aerobic condition. Here, it is possible to see that 21 genes have fold change higher or equal to 2.5. The gene that presents the highest value codifies for a NAD-dependent deacetylase which is an important regulator in cellular stress response and energy metabolism (Gou et at, 2012). The table also presents genes that are associated with ABC transporter proteins, chaperones, and others. On table 2, 27 genes have expression higher or equal to 2.5. The gene that has the highest value anaerobically codifies a protein associated with glutathione biosynthesis pathway. Other genes codify proteins important for ABC transporter, and cell wall synthesis. The two tables generated in this paper will be used as base for future research. 


```{r genes_large_anaero , echo=FALSE, warning=FALSE, message=FALSE}
library(knitr)

genes_large_anaero = read.csv('genes_filter_large_anaero.csv')
genes_large_anaero = genes_large_anaero %>% select(gene_description, fold_change_scaled) %>% mutate(fold_change_scaled = round(fold_change_scaled, 1)) %>% arrange(desc(fold_change_scaled)) %>% rename('Fold change'= fold_change_scaled, 'Protein' = gene_description) 

kable(genes_large_anaero, caption = 'Proteins coded by uperregulated genes and fold change (>= 2.5) in gene expression under anaerobic condition')

```



#Conclusion

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This paper presents results of initial analyzes of RNAseq data of *H. para* aerobically and anaerobically. It is possible to observe that 258 genes are upregulated in those conditions. Large part of the genes (170) are upregulated aerobically while 88 are upregulated anaerobically. Also, 48 genes are 2.5 times more expressed either aerobically or anaerobically. Future directions include more detailed analyzes of each of those 48 genes and, for some of them, construction of mutants that will be tested in specific conditions such as aerobically, anaerobically, and in co-cultures with other species that colonize the oral cavity to investigate the interaction between them. 

#References

Guo, X. Kesimer, M. Tolun, G. Zheng, X, Xu,Q. Lu,J. Sheehan, J.K. Griffith, J.D. Li,X. (2012). The NAD+-dependent protein deacetylase activity of SIRT1 is regulated by its oligomeric status. Scientific Reports volume 2, Article number: 640.

Park, O.J., Yi, H., Jeon, J. H., Kang, S.S., Koo, K.T., Kum, K.Y., . Han, S. H. (2015). Pyrosequencing Analysis of Subgingival Microbiota in Distinct Periodontal Conditions. Journal of Dental Research, 94(7), 921-927. https://doi.org/10.1177/0022034515583531.

Posse, J.L. Dios, P.D. Scully,C. *Saliva Protection and Transmissible Diseases*. Academic Press, 2018. Print.

